chromadb/api/collection_configuration.py

```diff
@@ -0,0 +1,278 @@
+        if "embedding_function" not in json_map:
+            return CollectionConfiguration(hnsw=json_to_hnsw_config(json_map["hnsw"]))
+        else:
+            if json_map["embedding_function"]["type"] == "legacy":
+                warnings.warn(
+                    "legacy embedding function config",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+                return CollectionConfiguration(
+                    hnsw=json_to_hnsw_config(json_map["hnsw"])
+                )
+            else:
+                ef = cast(
+                    EmbeddingFunction[Embeddable],
+                    known_embedding_functions[json_map["embedding_function"]["name"]],
+                )
+                return CollectionConfiguration(
+                    hnsw=json_to_hnsw_config(json_map["hnsw"]),
+                    embedding_function=ef.build_from_config(
+                        json_map["embedding_function"]["config"]
+                    ),
+                )
+
+
+class CreateHNSWConfiguration(TypedDict, total=False):
+    space: Space
+    ef_construction: int
+    max_neighbors: int
+    ef_search: int
+    num_threads: int
+    batch_size: int
+    sync_threshold: int
+    resize_factor: float
+
+
+class CreateCollectionConfiguration(TypedDict, total=False):
+    hnsw: Optional[CreateHNSWConfiguration]
+    embedding_function: Optional[EmbeddingFunction[Embeddable]]
+
+
+def load_collection_config_from_create_collection_config(
+    config: CreateCollectionConfiguration,
+) -> CollectionConfiguration:
+    return CollectionConfiguration(
+        hnsw=config.get("hnsw"), embedding_function=config.get("embedding_function")
+    )
+
+
+def create_collection_config_from_legacy_params(
+    metadata: Metadata,
+) -> CreateCollectionConfiguration:
+    """Create a CreateCollectionConfiguration from legacy parameters"""
+    old_to_new = {
+        "hnsw:space": "space",
+        "hnsw:ef_construction": "ef_construction",
+        "hnsw:max_neighbors": "max_neighbors",
+        "hnsw:ef_search": "ef_search",
+        "hnsw:num_threads": "num_threads",
+        "hnsw:batch_size": "batch_size",
+        "hnsw:sync_threshold": "sync_threshold",
+        "hnsw:resize_factor": "resize_factor",
+    }
+    json_map = {}
+    for name, value in metadata.items():
+        if name not in old_to_new:
+            raise ValueError(f"unknown legacy parameter: {name}")
+        json_map[old_to_new[name]] = value
+
+    hnsw_config = json_to_hnsw_config(json_map)
+    hnsw_config = populate_create_hnsw_defaults(hnsw_config)
+    validate_hnsw_config(hnsw_config)
+
+    return CreateCollectionConfiguration(hnsw=hnsw_config)
+
+
+def create_collection_config_to_json_str(config: CreateCollectionConfiguration) -> str:
+    """Convert a CreateCollection configuration to a JSON-serializable string"""
+    return json.dumps(create_collection_config_to_json(config))
+
+
+def create_collection_config_to_json(
```

Comment:
might be cleaner to phrase this as `CreateCollectionConfiguration -> CollectionConfiguration` instead